astroquery/alma/core.py

```diff
@@ -38,37 +191,60 @@ class AlmaClass(QueryWithLogin):
     USERNAME = conf.username
 
     def __init__(self):
+        # sia service does not need disambiguation but tap does
         super(AlmaClass, self).__init__()
+        self._sia = None
+        self._tap = None
+        self.sia_url = None
+        self.tap_url = None
 
-    def query_object_async(self, object_name, cache=True, public=True,
+    @property
+    def sia(self):
+        if not self._sia:
+            base_url = self._get_dataarchive_url()
+            if base_url.endswith('/'):
+                self.sia_url = base_url + ALMA_SIA_PATH
+            else:
+                self.sia_url = base_url + '/' + ALMA_SIA_PATH
+            self._sia = pyvo.dal.sia2.SIAService(baseurl=self.sia_url)
+        return self._sia
+
+    @property
+    def tap(self):
+        if not self._tap:
+            base_url = self._get_dataarchive_url()
+            if base_url.endswith('/'):
+                self.tap_url = base_url + ALMA_TAP_PATH
+            else:
+                self.tap_url = base_url + '/' + ALMA_TAP_PATH
+            self._tap = pyvo.dal.tap.TAPService(baseurl=self.tap_url)
+        return self._tap
+
+    def query_object_async(self, object_name, cache=None, public=True,
                            science=True, payload=None, **kwargs):
         """
-        Query the archive with a source name
+        Query the archive for a source name.
 
         Parameters
         ----------
         object_name : str
-            The object name.  Will be parsed by SESAME on the ALMA servers.
-        cache : bool
-            Cache the query?
+            The object name.  Will be resolved by astropy.coord.SkyCoord
+        cache : deprecated
```

Comment:
is it necessary to deprecate caching?  Does pyvo not support it?  The caching functionality is extremely useful, even though it is problematic in several ways (we have no way to test if the cache is out of date, for instance)